### ISSUE TITLE
flatpak

### ISSUE THREAD

**Issue by hebrewd** [antoyo/titanium#23]
Adding a flatpak mainfest for the repo will make the browser more accessible and then further down it can be available over at flathub.

Will you consider adding that? or accepting a pull request with the required changes?

**Comment by antoyo** [comment]
Which changes are required in titanium itself in order to support flatpack?

**Comment by hebrewd** [comment]
@antoyo I don't know of any that are needed within it, there is an additional file that needed for the repository, which is a flatpak mainfest.
It resembles a dockerfile for that matter, which is an instructions of what needed and how to build the flatpak that will hold titanium.

**Comment by antoyo** [comment]
As per the instructions [here](https://github.com/flathub/flathub/wiki/App-Submission) I would need to create a PR with the manifest.
Could you please write one so that I can submit it to flathub?

**Comment by hebrewd** [comment]
Yes, I will work on composing one now, I don't have much experience with generating one for rust projects, so it might not take an instant.

**Comment by hebrewd** [comment]
@antoyo So, I created a fork and added the necessary files. However, it seems like for the build to be compatible with flathub, the cargo packages need to be available as an archive somewhere.
(Via cargo vendor)

**Comment by antoyo** [comment]
I'm not sure how this works.
I thought you needed to fork the [flathub](https://github.com/flathub/flathub) directory in order to create a flatpak manifest.
What are the scripts you created used for?

**Comment by hebrewd** [comment]
Yes, that is true, however, flatpak doesn't support cargo as a build system, so I added a meson file to wrap it around.
The scripts are to allow an automatic build, so they can be built during the flatpak generation.

Then, you need the mainfest in addition to that.

**Comment by antoyo** [comment]
I don't really feal enthusiastic for maintaining build files where all files of the project should be listed.
Could the bash scripts call cargo directly instead of using meson?

**Comment by hebrewd** [comment]
To be in flathub those scripts will have to do the things that meson support, so technically yes, though implementing meson features in bash seems like a hassle.

To be honest, I am surprised at the complexity of this myself, I assumed it will be easier with rust, due to cargo, but it seems to be even more involved.

**Comment by ergpopler** [comment]
maybe an appimage or snap would be easier? They provide similar levels of ease to use and may be easier to make. AppImages are great.

**Comment by antoyo** [comment]
If that doesn't require maintaining a list of build files, I'd be happy to accept a PR if needed for that kind of package.
